src/sql-parser/src/parser.rs

```diff
@@ -903,20 +903,26 @@ impl<'a> Parser<'a> {
         let tok = self.next_token().unwrap(); // safe as EOF's precedence is the lowest
 
         let regular_binary_operator = match &tok {
-            Token::Op(s) => Some(s.as_str()),
-            Token::Eq => Some("="),
-            Token::Star => Some("*"),
-            Token::Keyword(ILIKE) => Some("~~*"),
-            Token::Keyword(LIKE) => Some("~~"),
+            Token::Op(s) => Some(Op::Bare(s.clone())),
+            Token::Eq => Some(Op::Bare(String::from("="))),
+            Token::Star => Some(Op::Bare(String::from("*"))),
+            Token::Keyword(ILIKE) => Some(Op::Bare(String::from("~~*"))),
+            Token::Keyword(LIKE) => Some(Op::Bare(String::from("~~"))),
             Token::Keyword(NOT) => {
                 if self.parse_keyword(LIKE) {
-                    Some("!~~")
+                    Some(Op::Bare(String::from("!~~")))
                 } else if self.parse_keyword(ILIKE) {
-                    Some("!~~*")
+                    Some(Op::Bare(String::from("!~~*")))
                 } else {
                     None
                 }
             }
+            Token::Keyword(OPERATOR) if self.peek_token() == Some(Token::LParen) => {
```

Comment:
it will consume any token with the text "operator" in it. Unfortunately I can't find the query that had an issue with it now. @petrosagg you reached the same conclusion in your PR, do you remember why it was required?